### PR TITLE
The blood dimension opens

### PR DESCRIPTION
**Pull request by Exo-212**
## About The Pull Request
This is my first PR! woo! I'm going to assume there will be workshopping of this, but here goes:

(Should) Adds two new reagents, Hydrofluoric Acid and Synthetic Thermal Solution. Synthetic Thermal Solution is NOT implemented as an exotic blood type at this time pending a full synth species being chosen/created/ported. Synthetic Thermal Solution has the appearance of very pale blue liquid. I added recipes for both the acid and the solution, so they should be able to be created by chemistry in round. 

Hydrofluoric Acid is a necessary ingredient reagent to synthetic thermal solution.

Hydrofluoric Acid is slightly more damaging than nitric acid, but is the same toxicity.

## Why It's Good For The Game
Unique blood for robots! Establishes a reagent with room to grow beyond its roots, unlike oil - which needs to remain oil for other recipes and use as fuel.

## Changelog

:cl:
add: synth_blood.dm
add: Hydrofluoric Acid (code found in synth_blood.dm)
/:cl:

### CODE REVIEW

File: modular_doppler/modular_medical/reagents/synth_blood.dm [hunk 1]
```diff
@@ -0,0 +1,82 @@
+// Reagent(s) to be used by synthetics and or androids when they are implemented as a selectable species.
+//I'm putting the recipes and precursor chems here for ease of use.
+/datum/reagent/synth_blood
+    name = "Synthetic Thermal Solution"
+    description = "A non-reactive liquid commonly utilized by humanoid synthetics to insulate their internal systems and conduct heat away from internal components."
+    taste_description = "thick water"
+    taste_mult = 1
+    color = "#A9FBFB"
+    reagent_state = LIQUID
+	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
```
Comment (Nerev4r) [inline modular_doppler/modular_medical/reagents/synth_blood.dm:10]: unindent here

File: modular_doppler/modular_medical/reagents/synth_blood.dm [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    mass = 10
+    color = "#AAAAAA77"
+    toxpwr = 1
+    acidpwr = 10.0
+    chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
+
+/datum/reagent/toxin/acid/hyflo_acid/on_mob_life(mob/living/carbon/affected_mob, seconds_per_tick, times_fired)
+    . = ..()
+    if(affected_mob.adjustFireLoss((volume/10) * REM * normalise_creation_purity() * seconds_per_tick, updating_health = FALSE, required_bodytype = affected_bodytype))
+    return UPDATE_MOB_HEALTH
```
Comment (Nerev4r) [inline modular_doppler/modular_medical/reagents/synth_blood.dm:58]: indent here

File: modular_doppler/modular_medical/reagents/synth_blood.dm [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+
+/datum/chemical_reaction/hyflo_acid
+    results = list(/datum/reagent/toxin/acid/hyflo_acid = 2)
+    required_reagents = list(/datum/reagent/hydrogen = 1, /datum/reagent/fluorine = 1)
+    required_catalyists = list(/datum/reagent/water = 5)
```
Comment (Nerev4r) [inline modular_doppler/modular_medical/reagents/synth_blood.dm]: catalysts

File: modular_doppler/modular_medical/reagents/synth_blood.dm [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    overheat_temp = 292
+    optimal_ph_min = 0
+    optimal_ph_max = 2
+    determin_ph_range = 5
+    temp_exponent_factor = 1
+    ph_exponent_factor = 10
+    thermic_constant = -200
+    H_ion_release = -20
+    rate_up_lim = 25
+    purity/min = 0
```
Comment (Nerev4r) [inline modular_doppler/modular_medical/reagents/synth_blood.dm:79]: purity_min

**Review by Nerev4r — DISMISSED** [review]

File: modular_doppler/modular_medical/reagents/synth_blood.dm (hunk 2 above)
Comment (Nerev4r) [reply modular_doppler/modular_medical/reagents/synth_blood.dm:57]: indent here!!!!!!!!!!!!!!!!

**Comment by Exo-212** [comment]
Screenshot 2024-10-03 131048
Screenshot 2024-10-03 131118
Screenshot 2024-10-03 131453

The system works

**Review by Nerev4r — APPROVED** [review]

**Pull request merged**